kartothek/core/factory.py

```diff
@@ -16,10 +17,17 @@ def _ensure_factory(
     factory: Optional["DatasetFactory"],
     load_dataset_metadata: bool,
     load_schema: bool = True,
+    gen_uuid_if_none: bool = False,
```

Comment:
The `DatasetFactory` still has certain design flaws and it was not intended to be used for writing paths. Primarily because it only makes sense if the dataset actually exists since it's effectively a caching layer. That's obviously not the case if we're in a pure write operation. I'm afraid that when using it for storage operations as well this leads to some weird bugs.

Reply:
Does it make sense to refactor this as part of a different issue?